crates/turborepo-lib/src/config/turbo_config.rs

```diff
@@ -0,0 +1,592 @@
+
+    // Process signature
+    let signature = if let Some(signature) = output_map.get("signature") {
+        match signature.as_str() {
+            "0" => Some(false),
+            "1" => Some(true),
+            _ => {
+                return Err(ConfigError::Anyhow(anyhow!(
+                    "TURBO_SIGNATURE should be either 1 or 0."
+                )))
+            }
+        }
+    } else {
+        None
+    };
+
+    // Process preflight
+    let preflight = if let Some(preflight) = output_map.get("preflight") {
+        match preflight.as_str() {
+            "0" => Some(false),
+            "1" => Some(true),
+            _ => {
+                return Err(ConfigError::Anyhow(anyhow!(
+                    "TURBO_PREFLIGHT should be either 1 or 0."
+                )))
+            }
+        }
+    } else {
+        None
+    };
+
+    // Process enabled
+    let enabled = if let Some(enabled) = output_map.get("enabled") {
+        match enabled.as_str() {
+            "0" => Some(false),
+            "1" => Some(true),
+            _ => {
+                return Err(ConfigError::Anyhow(anyhow!(
+                    "TURBO_REMOTE_CACHE_ENABLED should be either 1 or 0."
+                )))
+            }
+        }
+    } else {
+        None
+    };
+
+    // Process timeout
+    let timeout = if let Some(timeout) = output_map.get("timeout") {
+        Some(timeout.parse::<u64>().map_err(|e| {
+            ConfigError::Anyhow(anyhow!(
+                "TURBO_REMOTE_CACHE_TIMEOUT: error parsing timeout. {}",
+                e
+            ))
+        })?)
+    } else {
+        None
+    };
+
+    let output = ConfigurationOptions {
+        api_url: output_map.get("api_url").cloned(),
+        login_url: output_map.get("login_url").cloned(),
+        team_slug: output_map.get("team_slug").cloned(),
+        team_id: output_map.get("team_id").cloned(),
+        token: output_map.get("token").cloned(),
+
+        // Processed booleans
+        signature,
+        preflight,
+        enabled,
+
+        // Processed numbers
+        timeout,
+    };
+
+    Ok(output)
+}
+
+fn get_override_env_var_config(
+    environment: &HashMap<OsString, OsString>,
+) -> Result<ConfigurationOptions, ConfigError> {
+    let mut vercel_artifacts_mapping = HashMap::new();
+    vercel_artifacts_mapping.insert(OsString::from("vercel_artifacts_token"), "token");
+    vercel_artifacts_mapping.insert(OsString::from("vercel_artifacts_owner"), "team_id");
+
+    let mut output_map = HashMap::new();
+
+    // Process the VERCEL_ARTIFACTS_* next.
+    vercel_artifacts_mapping
+        .iter()
+        .try_for_each(|(k, mapped)| -> Result<(), ConfigError> {
+            if let Some(value) = environment.get(k) {
+                let converted = value.to_str().ok_or_else(|| {
+                    ConfigError::Anyhow(anyhow!(
+                        "{} is not UTF8.",
+                        k.to_ascii_uppercase().to_str().unwrap()
+                    ))
```

Comment:
Reading this freaked me out because we were unwrapping a key. We don't need to store the valid UTF8 keys as `OsString` in order to query the env:
```suggestion
            if let Some(value) = environment.get::<std::ffi::OsStr>(k.as_ref()) {
                let converted = value.to_str().ok_or_else(|| {
                    ConfigError::Anyhow(anyhow!(
                        "{} is not UTF8.",
                        k.to_ascii_uppercase()
                    ))
```

Reply:
Since it doesn't really matter at definition time if this is an `OsString` I chose that so we can call `std::env::vars_os()` which doesn't do an unnecessary UTF8 pass and (more-importantly) ergonomically doesn't require us to do any plumbing of errors we know won't occur.

The only thing that's weird here is that it calls unwrap on one of our own specified keys, which we can tell by inspection is safe (and we're not at risk of creating a mapping that isn't UTF8).

Reply:
https://github.com/vercel/turbo/pull/5796/commits/49da009ca2caa38f8b26627b59c22928f5407daf